src/WebJobs.Script/Host/FunctionMetadataProvider.cs

```diff
@@ -16,22 +17,24 @@ internal class FunctionMetadataProvider : IFunctionMetadataProvider
     {
         private readonly IEnvironment _environment;
         private readonly ILogger<FunctionMetadataProvider> _logger;
+        private readonly FunctionsHostingConfigOptions _functionsHostingConfigOptions;
         private IWorkerFunctionMetadataProvider _workerFunctionMetadataProvider;
         private IHostFunctionMetadataProvider _hostFunctionMetadataProvider;
 
-        public FunctionMetadataProvider(ILogger<FunctionMetadataProvider> logger, IWorkerFunctionMetadataProvider workerFunctionMetadataProvider, IHostFunctionMetadataProvider hostFunctionMetadataProvider)
+        public FunctionMetadataProvider(ILogger<FunctionMetadataProvider> logger, IWorkerFunctionMetadataProvider workerFunctionMetadataProvider, IHostFunctionMetadataProvider hostFunctionMetadataProvider, IOptionsMonitor<FunctionsHostingConfigOptions> functionsHostingConfigOptions)
         {
             _logger = logger;
             _workerFunctionMetadataProvider = workerFunctionMetadataProvider;
             _hostFunctionMetadataProvider = hostFunctionMetadataProvider;
+            _functionsHostingConfigOptions = functionsHostingConfigOptions.CurrentValue;
```

Comment:
Is there a reason we're taking a dependency on `IOptionsMonitor` here if we're not observing changes? Is this to avoid additional setup runs?

Reply:
Not really, have switched to ioptions